drake/systems/plants/RigidBodySystem.h

```diff
@@ -237,6 +295,51 @@ namespace Drake {
   public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
   };
+
+  /** RigidBodySensor
+   * @brief interface class for elements which define a sensor which reads the state of a rigid body system
+   */
+  class DRAKERBSYSTEM_EXPORT RigidBodySensor {
+  public:
+    RigidBodySensor(RigidBodySystem* sys, const std::string& name) : sys(sys), name(name) {}
+    virtual ~RigidBodySensor() {}
+
+    virtual size_t getNumOutputs() const { return 0; }
+    virtual Eigen::VectorXd output(const double& t, const KinematicsCache<double>& rigid_body_state) const = 0;
+
+  protected:
+    RigidBodySystem* sys;
```

Comment:
Noticed a lot of raw pointers around RigidBodySystem.  Is this for compatibility with legacy code?  Why not smart pointers?